android/src/toga_android/widgets/label.py

```diff
@@ -78,4 +72,4 @@ def rehint(self):
         self.interface.intrinsic.width = at_least(self.native.getMeasuredWidth())
 
     def set_alignment(self, value):
-        self.set_textview_alignment(value, Gravity.CENTER_VERTICAL)
```

Comment:
Because of the getLayoutParams issue, this line was never previously taking effect during widget creation, and the label remained at the system default alignment of TOP.

After adding a label with an explicit height to the `resize` example, it turns out the vertical alignment of the text is inconsistent between the backends:
* TOP: Android, Cocoa
* CENTER: GTK, iOS, Winforms

In the interest of staying as close as possible to CSS, I suggest we standardize on TOP.

Reply:
Agreed; although we probably want to investigate making this configurable at some point (possibly part of the discussion around #1194). Centered vertical alignment is a very useful option for GUI layout, as it means the centreline of labels is aligned with the centreline of text inputs, widgets, etc.

Actually making the label vertically aligned was trivial on GTK... and more complex than it should be for iOS.

Reply:
Propagating this through to TextInput, it gets a bit more complicated. There's a similar degree of inconsistency about where the text appears in the input... but there's also a near complete lack of control over this feature on macOS, iOS and GTK. macOS uses top alignment by default; iOS and GTK use vertically centered. FWIW, HTML (Safari on macOS) uses vertically centered alignment, but AFAICT there's no way to style this with CSS.

However, IMHO, in the case of TextInput, this is getting to the point where we're overriding platform defaults. In the case of label, it makes sense to have a consistent vertical alignment because we're rendering the location of text in a larger layout, there's no "external" chrome, and setting a height on a label isn't fundamentally unreasonable. But the external chrome of a TextInput will fill the entire allocated space, and should only have a single line of content. Setting a height on a TextInput is generally ill-advised anyway, as a text input should generally be the height of the line of text and no more. I'd argue that if the user is setting an explicit height, it's the operating system's decision where that line of text should appear. 

*If* the platform provides API control for vertical alignment of the text in a TextInput, we can pick a preferred alignment (Center to match HTML?); but if there's no explicit control, then I don't think we should go to extraordinary measures to override the default behavior. 

I think a similar argument could be made for MultilineTextInput - with a more obvious default being TOP. However, I think Android might be the only platform affected by this, and mostly because every widget in Android is apparently a TextLabel :-) Every other platform seems to use TOP for multilinetext, without being configurable.

Reply:
Agreed: I think MultilineTextInput should definitely be TOP aligned, but setting the height of the single-line TextInput doesn't really make any sense, so it's fine for the alignment of that to be undefined.